### PR TITLE
Add CI build rules to CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by psibi**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -160,6 +160,34 @@ Where again, `<PATTERN>` is the name of the folder listed in the
...
 
+## CI Build rules
+
+We use [Azure](https://dev.azure.com/commercialhaskell/stack/_build)
+to do CI builds on Stack. There are two types of build which happens
+there:
+
+### Unit based build
+
+This build runs only the unit tests in the Stack codebase. These are
```
Comment (snoyberg) [inline CONTRIBUTING.md]: It does more than that: it builds the code with `--pedantic`, it runs hlint checks, _and_ it runs all test suites on multiple GHC/OS configurations.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -160,6 +160,34 @@ Where again, `<PATTERN>` is the name of the folder listed in the
...
+* CI will run this for any branches starting with `ci/`
+* CI will run this for all new PR's.
+
+### Integration based build
+
+This build runs the integration tests in the Stack codebase. This is
+scheduled to run daily once for both the stable and master branches.
+
+Also, you can manually run this on a specific branch from the Azure UI
+if you have the appropriate permissions.
```
Comment (snoyberg) [inline CONTRIBUTING.md]: Perhaps comment: if you'd specifically like a branch or PR to run integration tests, add a comment in the PR and we can queue one up.

Comment (psibi) [reply CONTRIBUTING.md]: Added them.

File: CONTRIBUTING.md (hunk 1 above)
Comment (psibi) [reply CONTRIBUTING.md]: I have added that info now.

**Pull request merged**